### PR TITLE
Set PRJDIR to default if not provided by vars.json

### PR DESCRIPTION
**Pull request by aaannz**
if PRJDIR is not provided by vars.json try to set it to default
'/var/lib/openqa/share'. But do so only if CASEDIR is subdir of
this directory, otherwise report error and bail out

See https://progress.opensuse.org/issues/16362

### CODE REVIEW

**Comment by aaannz** [comment]
@okurz WDYT

File: bmwqemu.pm [hunk 1]
```diff
@@ -153,7 +162,7 @@ sub init {
     ## env vars end
 
     ## some var checks
-    if (!-x $gocrbin) {
+    if ($gocrbin && !-x $gocrbin) {
```
Comment (okurz) [inline bmwqemu.pm]: unrelated. Please move to different commit (same PR is fine)

File: bmwqemu.pm [hunk 2]
```diff
@@ -126,7 +126,7 @@ sub init {
 
     unless ($vars{CASEDIR}) {
         die "DISTRI undefined\n" . pp(\%vars) . "\n" unless $vars{DISTRI};
-        my @dirs = ("$scriptdir/distri/$vars{DISTRI}");
+        my @dirs = ("$scriptdir/tests/$vars{DISTRI}");
```
Comment (okurz) [inline bmwqemu.pm]: I don't get the meaning of "tests/" dir. Is this also only a nice "default here?

File: t/12-bmwqemu.t [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    open($varsfh, '<', 'vars.json') || BAIL_OUT('can not open vars.json for reading');
+    my $ret;
+    eval { $ret = JSON->new->relaxed->decode(<$varsfh>); };
+    die "parse error in vars.json:\n$@" if $@;
+    close($varsfh);
+    my %vars = %{$ret};
+    ok(!$vars{DISTRI}, 'DISTRI not supplied and not set');
+    is($vars{CASEDIR}, $dir, 'CASEDIR unchanged');
+    ok(!$vars{PRJDIR}, 'PRJDIR not supplied and not set');
+};
```
Comment (okurz) [inline t/12-bmwqemu.t]: that test has a lot of duplication. Makes hard to see what are the differences among the three subtests. Can you extra helper function(s)?

File: bmwqemu.pm (hunk 2 above)
Comment (aaannz) [inline bmwqemu.pm]: I don't get this check at all. We don't have `distri` directory AFAIK, but use tests instead. So this test IMO never passed. `$striptdir` is also set to some `/usr/lib/os-autoinst` instead of `/var/lib/openqa[/share]`. I'll remove this part and consider `CASEDIR` to be mandatory.

**Comment by aaannz** [comment]
Updated and added one more commit to remove obsolete hack

**Comment by okurz** [comment]
LGTM, more DRY in test would be good but ok

**Comment by aaannz** [comment]
I did consider moving those `eval` blocks to sub, but I really didn't want to hide where the actual test is happening and what is being done.

**Comment by okurz** [comment]
LGTM then
